extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -514,6 +542,32 @@ private static OidcConfigurationMetadata createLocalMetadata(OidcTenantConfig oi
                 oidcConfig.token.issuer.orElse(null));
     }
 
+    private static void fireOidcServerNotAvailableEvent(String authServerUrl, String tenantId) {
+        if (fireOidcServerEvent(authServerUrl, OIDC_SERVER_NOT_AVAILABLE)) {
+            tenantsExpectingServerAvailableEvents.add(tenantId);
+        }
+    }
+
+    private static void fireOidcServerAvailableEvent(String authServerUrl, String tenantId) {
+        if (fireOidcServerEvent(authServerUrl, OIDC_SERVER_AVAILABLE)) {
+            tenantsExpectingServerAvailableEvents.remove(tenantId);
+        }
+    }
+
+    private static boolean shouldFireOidcServerAvailableEvent(String tenantId) {
+        return tenantsExpectingServerAvailableEvents.contains(tenantId);
```

Comment:
@michalvavrik This is the last one I think, that should probably remove the tenant from the map, I guess, otherwise, when a dynamic TenantConfigResolver is used, it might keep growing even if only when connection failures occur

Reply:
@sberyozkin I'll need little more to understand Sergey. Can you give me hint what you mean, please? Please note that tenant is removed when OIDC available event is reported. I don't want to remove it before that happens, because then I wouldn't have certainty that event were reported.